src/palindrom.js

```diff
@@ -193,319 +187,317 @@ var Palindrom = (function() {
   }
 
   function NoHeartbeat() {
-    this.start = this.stop = this.notifySend = this.notifyReceive = function() {};
+    this.start = this.stop = this.notifySend = this.notifyReceive = () => {};
   }
 
-  function PalindromNetworkChannel(
-    palindrom,
-    remoteUrl,
-    useWebSocket,
-    onReceive,
-    onSend,
-    onConnectionError,
-    onSocketOpened,
-    onFatalError,
-    onStateChange
-  ) {
-    // TODO(tomalec): to be removed once we will achieve better separation of concerns
-    this.palindrom = palindrom;
-
-    if (typeof window !== 'undefined' && window.location) {
-      this.remoteUrl = new URL(remoteUrl, window.location.href);
-    } else {
-      // in Node, URL is absolute
-      this.remoteUrl = new URL(remoteUrl);
-    }
-
-    onReceive && (this.onReceive = onReceive);
-    onSend && (this.onSend = onSend);
-    onConnectionError && (this.onConnectionError = onConnectionError);
-    onFatalError && (this.onFatalError = onFatalError);
-    onStateChange && (this.onStateChange = onStateChange);
-    onSocketOpened && (this.onSocketOpened = onSocketOpened);
-    //useWebSocket = useWebSocket || false;
-    var that = this;
-    Object.defineProperty(this, 'useWebSocket', {
-      get: function() {
-        return useWebSocket;
-      },
-      set: function(newValue) {
-        useWebSocket = newValue;
-
-        if (newValue == false) {
-          if (that._ws) {
-            that._ws.onclose = function() {
-              //overwrites the previous onclose
-              that._ws = null;
-            };
-            that._ws.close();
-          }
-          // define wsUrl if needed
-        } else if (!that.wsUrl) {
-          that.wsUrl = toWebSocketURL(that.remoteUrl.href);
+  class PalindromNetworkChannel {
+    constructor(
+      palindrom,
+      remoteUrl,
+      useWebSocket,
+      onReceive,
+      onSend,
+      onConnectionError,
+      onSocketOpened,
+      onFatalError,
+      onStateChange
+    ) {
+      // TODO(tomalec): to be removed once we will achieve better separation of concerns
+      this.palindrom = palindrom;
+
+      if (typeof window !== 'undefined' && window.location) {
+        this.remoteUrl = new URL(remoteUrl, window.location.href);
+      } else {
+        // in Node, URL is absolute
+        this.remoteUrl = new URL(remoteUrl);
+      }
+
+      onReceive && (this.onReceive = onReceive);
+      onSend && (this.onSend = onSend);
+      onConnectionError && (this.onConnectionError = onConnectionError);
+      onFatalError && (this.onFatalError = onFatalError);
+      onStateChange && (this.onStateChange = onStateChange);
+      onSocketOpened && (this.onSocketOpened = onSocketOpened);
+      this._useWebSocket = useWebSocket || false;
+    }
+    get useWebSocket() {
+      return this._useWebSocket;
```

Comment:
I'd move it back to the constructor, not to expose `_useWebSocket` to public.

Reply:
Done.